extension/chrome/settings/setup/setup-create-key.ts

```diff
@@ -66,7 +67,7 @@ export class SetupCreateKeyModule {
     await Settings.forbidAndRefreshPageIfCannot('CREATE_KEYS', this.view.rules!);
     const { full_name } = await Store.getAcct(this.view.acctEmail, ['full_name']);
     try {
-      const key = await PgpKey.create([{ name: full_name || '', email: this.view.acctEmail }], 'rsa4096', options.passphrase); // todo - add all addresses?
+      const key = await PgpKey.create([{ name: full_name || '', email: this.view.acctEmail }], options.key_type, options.passphrase); // todo - add all addresses?
```

Comment:
Since it's only needed for creating keys, you could probably grab the value from DOM either in this function, or add KeyAlgo as a second argument to this method, instead of including it in `SetupOptions`. If you think it's indeed better to keep it in SetupOptions, then you'd have to make it optional (needed only when creating).

The cleanest would probably be to change method sig to `public createSaveKeyPair = async (options: SetupOptions, keyAlgo: KeyAlgo)`

Reply:
The cleanest also seems to be the most simple and readable one, done :+1: